posthog/api/test/test_capture.py

```diff
@@ -129,7 +147,7 @@ def test_capture_event(self, kafka_produce):
                 ],
             },
         }
-        with self.assertNumQueries(1):
+        with self.assertNumQueries(2):
```

Comment:
If we call this again does it reduce to 1 query?

Reply:
It should, the test was getting too long so I didn't want to add another call here. I'll add a new test especifically for caching testing.

Reply:
Just pushed a test for this scenario: I'm firing off 20 requests in a row and only the first one is making two queries.

Reply:
Still have to assert the behavior under a lot of load (and look into the Redis client you linked in a different comment), but at least the "happy path" appears to be working fine :+1: